spec/04-basic-declarations-and-definitions.md

```diff
@@ -840,6 +840,23 @@ class C extends I {
 Here, it is OK to leave out the result type of `factorial`
 in `C`, even though the method is recursive.
 
+### Tail-Recursive Call Elimination
+
+Method definitions which contain self-recursive invocations in tail position
+are optimized for stack safety. Self-invocations which are the last operation
+before returning from the method are replaced with jumps to the beginning of
+the method, much as in a while loop. Sibling-invocations are also optimized.
```

Comment:
I think it needs to be explained what a "sibling-invocation" is.

(Wild guess: is it when `this` changes...? It took me some pondering to even formulate the guess.)

Reply:
I studied Wikipedia for usages slash nomenclature. https://en.wikipedia.org/wiki/Tail_call